pkg/controller/cloneset/sync/cloneset_scale.go

```diff
@@ -278,6 +290,17 @@ func (r *realControl) deletePods(cs *appsv1alpha1.CloneSet, podsToDelete []*v1.P
 				modified = true
 				clonesetutils.ResourceVersionExpectations.Expect(gotPod)
 			}
+
+			if lifecycle.IsPodNotReadyDuringPreparingDeletePolicy(cs.Spec.Lifecycle) {
+				if updated, err := podreadiness.SetPodNotReadyDuringPreparingDelete(r.Client, pod, getReadinessMessage(cs)); err != nil {
+					return false, err
+				} else if updated {
+					klog.V(3).Infof("CloneSet %s scaling update pod %s not ready condition due to PreparingDelete",
+						clonesetutils.GetControllerKey(cs), pod.Name)
+					modified = true
```

Comment:
All these `modified` are unnecessary. We only use the `modified` set by `UpdatePodLifecycle` before with `ResourceVersionExpectations` to avoid duplicate operation in next reconciles.